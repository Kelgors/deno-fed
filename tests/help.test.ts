import { expect } from '@std/expect';
import { describe, it } from '@std/testing/bdd';
import { assertSnapshot } from '@std/testing/snapshot';
import { fed } from './utils/exec.ts';

describe('help', () => {
  it('should display the cli usage', async (t) => {
    const { stdout, exitCode } = await fed(['--help']);
    expect(exitCode).toBe(0);
    assertSnapshot(t, stdout);
  });
});
