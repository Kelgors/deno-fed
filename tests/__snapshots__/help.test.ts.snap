// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`help > should display the cli usage 1`] = `
[
  "Usage: cli [options]",
  "",
  "Iterate over directories",
  "",
  "Options:",
  "  -v, --version                 Display CLI version (default: false)",
  "  --verbose                     Display browsed folders",
  "  --cwd <PATH>                  Set the working directory",
  "  --git                         List only git repositories",
  "  --no-git                      List only non git repositories",
  "  --diff                        List git repositories with a current diff",
  "  --no-diff                     List git repositories without a diff",
  "  --detached                    List git repositories detached from a branch",
  "  --no-detached                 List git repositories currenly on a branch",
  "  --unpushed                    List git repositories containing commits not",
  "                                pushed to remote on the current branch",
  "  --no-unpushed                 List git repositories not containing commits",
  "                                not pushed to remote on the current branch",
  "  --stashes                     List git repositories with stash in it",
  "  --no-stashes                  List git repositories without stash in it",
  "  --with-file <FILE...>         Check file presence (default: [])",
  "  --without-file <FILE...>      Check file absence (default: [])",
  "  --node                        List only node projects",
  "  --no-node                     List only not node projects",
  "  --match <REGEX>               List only directory names matching the regex",
  "  --include-branch <BRANCH...>   (default: [])",
  "  --exclude-branch <BRANCH...>   (default: [])",
  "  --ignore-failures             Don't stop the program when error occurs",
  "                                (default: false)",
  "  -x, --exec <cmd>              run command for each directories (available",
  "                                variables {}, {path}, {diff}, {branch})",
  "                                (default: "echo {}")",
  "  -h, --help                    display help for command",
]
`;
